File: app/src/main/java/com/example/paetz/yacguide/database/Rock.java
package com.example.paetz.yacguide.database;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class Rock {

    // This needs to be in sync with sandsteinklettern.de!
    public static final char typeSummit = 'G';
    public static final char typeMassif = 'M';
    public static final char typeBoulder = 'B';
    public static final char typeStonePit = 'S';
    public static final char typeAlpine = 'A';
    public static final char typeCave = 'H';
    public static final char typeUnofficial = 'N';

    public static final char statusTemporarilyProhibited = 'Z';
    public static final char statusProhibited = 'X';
    public static final char statusPartlyProhibited = 'T';

    @PrimaryKey
    private int id;

    private float nr;
    private char type;   // 'G' = Rock, 'M' = Massif, 'H' = Cage, 'B' = Boulder, 'N' = unofficial
    private char status; // 'X' = prohibited, 'Z' = temporarily prohibited, 'T' = partly prohibited
    private String name;
    private float longitude;
    private float latitude;
    private boolean ascended;
    private int parentId;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public float getNr() {
        return nr;
    }

    public void setNr(float nr) {
        this.nr = nr;
    }

    public char getType() {
        return type;
    }

    public void setType(char type) {
        this.type = type;
    }

    public char getStatus() {
        return status;
    }

    public void setStatus(char status) {
        this.status = status;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public float getLongitude() {
        return longitude;
    }

    public void setLongitude(float longitude) {
        this.longitude = longitude;
    }

    public float getLatitude() {
        return latitude;
    }

    public void setLatitude(float latitude) {
        this.latitude = latitude;
    }

    public boolean getAscended() {
        return ascended;
    }

    public void setAscended(boolean ascended) {
        this.ascended = ascended;
    }

    public int getParentId() {
        return parentId;
    }

    public void setParentId(int parentId) {
        this.parentId = parentId;
    }
}

File: app/src/main/java/com/yacgroup/yacguide/database/Rock.kt
/*
 * Copyright (C) 2019, 2022 Axel Paetzold
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.yacgroup.yacguide.database

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Rock(
    @PrimaryKey val id: Int,
    val nr: Float,
    val type: Char,
    val status: Char,
    val name: String?,
    val longitude: Float,
    val latitude: Float,
    var ascendsBitMask: Int,    // a 32bit number for 14 ascend styles.
                                // Bit index == (styleId - 1).
                                // Bit value == 1 if rock has been ascended with this style.
                                // Note: styleId == 0 ("Unknown") corresponds to mask 0b10000...0 = -IntMax.
    val parentId: Int
) {
    companion object {

        // This needs to be in sync with sandsteinklettern.de!
        const val typeSummit = 'G'
        const val typeMassif = 'M'
        const val typeBoulder = 'B'
        const val typeStonePit = 'S'
        const val typeAlpine = 'A'
        const val typeCave = 'H'
        const val typeUnofficial = 'N'

        const val statusTemporarilyProhibited = 'Z'
        const val statusProhibited = 'X'
        const val statusPartlyProhibited = 'T'
        const val statusCollapsed = 'E'
    }
}
